### PR TITLE
Add caching and cach purging logic

### PR DESCRIPTION
**Pull request by phamann**
- Set a default cache-control header of `private, no-cache` for all requests
- Override default cache-control with CDN outbound rules for idempotent read requests `GET, HEAD`
- Send `PURGE` requests to Fastly on successful DB writes via `POST, PUT, DELETE`
- Add tests for above logic

/cc @Financial-Times/next-myft

### CODE REVIEW

File: server/lib/purge-cache.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+module.exports = function(actor) {
+	if(process.env.NODE_ENV !== 'test' && (!process.env.FASTLY_API_KEY || !process.env.FASTLY_SERVICE_ID)) {
```
Comment (adgad) [inline server/lib/purge-cache.js]: Should this have a !== 'production' too ?

Comment (phamann) [inline server/lib/purge-cache.js]: Production will always have the two keys on the right hand side of the conditional.

**Comment by adgad** [comment]
:+1:

File: server/lib/purge-cache.js (hunk 1 above)
Comment (wheresrhys) [inline server/lib/purge-cache.js]: d'you mean it should be  `=== 'production'` instead of `!=== 'test'` ?

Comment (phamann) [inline server/lib/purge-cache.js]: @wheresrhys That would simplify the logic yes :+1:

**Pull request merged**